src/Calendar/Calendar.php

```diff
@@ -472,8 +482,8 @@ private function deriveStatusTypeFromSubEvents(): StatusType
         $statusTypeCounts[StatusType::TemporarilyUnavailable()->toString()] = 0;
         $statusTypeCounts[StatusType::Unavailable()->toString()] = 0;
 
-        foreach ($this->timestamps as $timestamp) {
-            ++$statusTypeCounts[$timestamp->getStatus()->getType()->toString()];
+        foreach ($this->subEvents as $subEvent) {
```

Comment:
This seems over complicated. We just want to know if any status is of the type "Available", the moment we have one we can return the StatusType::Available().
No need to count how many there are.

You cannot do this for the other events because of the priority, but even there you don't need a counter, just a simple boolean.

Reply:
But how would you handle `TemporarilyUnavailable`, `Unavailable` and missing states?